policyengine_us/variables/gov/states/mt/tax/income/mt_capital_gains_tax_indiv.py

```diff
@@ -0,0 +1,80 @@
+    defined_for = StateCode.MT
+
+    def formula(person, period, parameters):
+        p = parameters(period).gov.states.mt.tax.income.main.capital_gains
+        # the tax for capital gains comes into effect after 2024
+        if p.in_effect:
+            capital_gains = person("long_term_capital_gains", period)
+            taxable_income = person("mt_taxable_income_indiv", period)
+            filing_status = person.tax_unit(
+                "state_filing_status_if_married_filing_separately_on_same_return",
+                period,
+            )
+            non_qualified_income = max_(taxable_income - capital_gains, 0)
+            threshold = p.threshold[filing_status]
+            non_qualified_income_exceeds_threshold = (
+                non_qualified_income > threshold
+            )
+            # When the nonqualified income exceeds the threshold, apply the main rate
+            capital_gains_main_tax = (
+                p.rates.main[filing_status] * capital_gains
+            )
+            lower_threshold = max_(
+                p.threshold[filing_status] - non_qualified_income, 0
+            )
+            status = filing_status.possible_values
+            base_rate = select(
+                [
+                    filing_status == status.SINGLE,
+                    filing_status == status.SEPARATE,
+                    filing_status == status.SURVIVING_SPOUSE,
+                    filing_status == status.HEAD_OF_HOUSEHOLD,
+                ],
+                [
+                    p.rates.single.calc(0),
+                    p.rates.separate.calc(0),
+                    p.rates.surviving_spouse.calc(0),
+                    p.rates.head_of_household.calc(0),
+                ],
+            )
+            lower_rate = select(
+                [
+                    filing_status == status.SINGLE,
+                    filing_status == status.SEPARATE,
+                    filing_status == status.SURVIVING_SPOUSE,
+                    filing_status == status.HEAD_OF_HOUSEHOLD,
+                ],
+                [
+                    p.rates.lower.single,
+                    p.rates.lower.separate,
+                    p.rates.lower.surviving_spouse,
+                    p.rates.lower.head_of_household,
+                ],
+            )
+            base_capital_gains_tax = base_rate * lower_threshold
+            lower_capital_gains_tax = where(
+                capital_gains <= lower_threshold,
+                capital_gains
+                * base_rate,  # First net long-term capital gians that is within the difference between the threshold and nonqualified income
+                base_capital_gains_tax
+                + (capital_gains - lower_threshold)
+                * lower_rate,  # Next net long-term capital gains that exceed the threshold minus nonqualified income
+            )
+
+            return where(
```

Comment:
could you break this up into the 3 variables in this where?

Reply:
I was trying to separate as much as possible but it would result in extensive code duplication as we need to re calculate the rates for each of the base on lower tax rate - I believe the current code is as much as I can split it up without duplicating the select statements

Reply:
I have consolidate as much as possible IMO, open for suggestions